test/lib/warn-upload-overwrite.spec.js

```diff
@@ -23,12 +25,14 @@ describe('warn-upload-overwrite', () => {
       calls.push(args);
     };
     sinon.stub(fs, 'exists').returns(true);
-    api.start();
+    apiStub.start();
   });
 
   afterEach(() => {
     sinon.restore();
-    return api.stop();
+    api = rewire('../../src/lib/api');
+    warnUploadOverwrite = rewire('../../src/lib/warn-upload-overwrite');
```

Comment:
That's the fun part of my latest problems with cht-conf tests...

In https://github.com/medic/cht-conf/blob/575-retry-on-failure/src/lib/api.js#L9 you will notice this on line 9 `const cache = new Map();`. 
This map basically caches some common responses from the API that can be reused between cht-conf commands like how some things in couch are configured. It took me a while to figure this out but this `cache` variable wouldn't get reset across tests, the map was still populated with the API response the first test had cached and it was causing tests in this file to fail.  
They were not failing before because the API response mocks were incomplete and it didn't matter much because the command could keep going without this information. But with the retry mechanism, a lack of API response means it gets retried a couple times before moving on and that made a lot of tests run over the timeout.

Back to your question, we need to rewire these because we need to reset the `cache` variable. I tried with just `api.__set__('cache', new Map());` in the `beforeEach` but it wasn't enough to reset it. So we rewire `api` and then we rewire `warnUploadOverwrite` to give it the newly rewired `api`.